narwhals/dataframe.py

```diff
@@ -449,7 +449,13 @@ def get_column(self, name: str) -> Series:
         )
 
     @overload
-    def __getitem__(self, item: tuple[Sequence[int], str | int]) -> Series: ...  # type: ignore[overload-overlap]
+    def __getitem__(self, item: tuple[Sequence[int], Sequence[int]]) -> Self: ...
+    @overload
+    def __getitem__(self, item: tuple[Sequence[int], str]) -> Series: ...  # type: ignore[overload-overlap]
+    @overload
+    def __getitem__(self, item: tuple[Sequence[int], Sequence[str]]) -> Self: ...
+    @overload
+    def __getitem__(self, item: tuple[Sequence[int], int]) -> Series: ...  # type: ignore[overload-overlap]
```

Comment:
this required some overload ignores, but overall, it looks correct:
```
dfpl = pl.DataFrame({'a': [1,2,3], 'b': [4,5,6]})

df = nw.from_native(dfpl, eager_only=True)

reveal_type(df[[0, 1], 'a'])
reveal_type(df[[0, 1], 0])
reveal_type(df[[0, 1], [0, 1]])
reveal_type(df[[0, 1], ['a', 'b']])
```
mypy reveals the first two to be Series, and the last two to be DataFrame